subprojects/build-cache/src/integTest/groovy/org/gradle/caching/configuration/BuildCacheConfigurationIntegrationTest.groovy

```diff
@@ -269,6 +275,37 @@ class BuildCacheConfigurationIntegrationTest extends AbstractIntegrationSpec {
         localBuildCache.empty
     }
 
+    @Unroll
+    def "shows deprecation warning when using custom local cache using #customCacheConfig"() {
+        settingsFile << """
+            class CustomBuildCache extends AbstractBuildCache {}
+            class CustomBuildCacheFactory implements BuildCacheServiceFactory<CustomBuildCache> {
+                @Override BuildCacheService createBuildCacheService(CustomBuildCache configuration, Describer describer) { 
+                    throw new UnsupportedOperationException() 
+                }
+            }
+            
+            buildCache {
+                registerBuildCacheService(CustomBuildCache, CustomBuildCacheFactory)
+                
+                $customCacheConfig
+            }
+        """
+        executer.expectDeprecationWarning()
+
+        when:
+        run "help"
+
+        then:
+        output.contains("Using a local cache type other than DirectoryBuildCache has been deprecated. This is scheduled to be removed in Gradle 6.0.")
```

Comment:
Shall we use `build cache` in the message instead of only `cache`?

Reply:
Fixed.